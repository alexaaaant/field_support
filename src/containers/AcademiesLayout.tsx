import React from 'react';
import { AcademiesInfo } from '../components/AcademyInfo';
import { AcademiesList } from '../components/AcademyList';

export const AcademiesLayout: React.FC = () => {
    return (
        <div className="layout">
            <div className="layout__main">
                <AcademiesList />
            </div>
            <div className="layout__extra">
                <AcademiesInfo />
            </div>
        </div>
    );
};