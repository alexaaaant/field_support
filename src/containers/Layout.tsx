import React from 'react';
import '../styles/layout.css';
import { AcademiesInfo } from '../components/AcademiesInfo';
import { AcademiesList } from '../components/AcademiesList';

export const Layout: React.FC = () => {
    return (
        <div className="layout">
            <div className="layout__main">
                <AcademiesList />
            </div>
            <div className="layout__extra">
                <AcademiesInfo />
            </div>
        </div>
    );
};