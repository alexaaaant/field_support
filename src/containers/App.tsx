import React, { useEffect, useState } from 'react';
import { useDispatch } from '../common/hooks/useDispatch';
import { getBatteries } from '../services/api';
import { academiesAdded } from '../slices/academiesSlice';
import { AcademiesLayout } from './AcademiesLayout';

export const App: React.FC = () => {
    const [isLoading, setIsLoading] = useState(true);
    const dispatch = useDispatch();

    useEffect(() => {
        const fetchBatteries = async () => {
            const data = await getBatteries();
            dispatch(academiesAdded(data));
            setIsLoading(false);
        };
        fetchBatteries();
    // eslint-disable-next-line react-hooks/exhaustive-deps
    },        []);
       
    return (
        <div>{isLoading ? 
        <div>Loading</div> :
        <AcademiesLayout />
    }</div>
    );
};