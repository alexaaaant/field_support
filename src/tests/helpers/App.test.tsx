import { act } from 'react-dom/test-utils';
import { App } from '../../containers/App';
import { fireEvent, flushPromises, render, screen } from './test-utils';
import batteriesRaw from '../../data/battery-data.json';

const mockedGetBatteries = () => Promise.resolve(batteriesRaw);

jest.useFakeTimers();
jest.mock('../../services/api', () => {
    const original = jest.requireActual('../../services/api');

    return {
        ...original,
        getBatteries: () => mockedGetBatteries(),
    };
});

describe('App', () => {
    it('should display a spinner before data', async() => {
        const { container } = render(<App />);

        expect(container).toMatchSnapshot();
        
        await act(async () => {
            jest.runAllTimers();
            await flushPromises();
        });

        expect(container).toMatchSnapshot();
    });

    it('should display an academy info after click on it', async() => {
        const { container } = render(<App />);

        await act(async () => {
            jest.runAllTimers();
            await flushPromises();
        });

        const academyElement = screen.getByText('30013');
        fireEvent.click(academyElement);

        expect(container).toMatchSnapshot();
    });
});