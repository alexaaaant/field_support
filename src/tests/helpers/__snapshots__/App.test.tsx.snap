// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should display a spinner before data 1`] = `
<div>
   
  <div
    class="container"
  >
    <div>
      Loading...
    </div>
  </div>
</div>
`;

exports[`App should display a spinner before data 2`] = `
<div>
   
  <div
    class="container"
  >
    <div
      class="layout"
    >
      <div
        class="layout__main"
      >
        <div
          class="academies-list"
        >
          <span
            class="header"
          >
            Academies ID:
          </span>
          <div
            class="academies-list__container"
          >
            <div
              class="academies-list__academy "
              data-academy="true"
              data-id="30006"
            >
              30006
            </div>
            <div
              class="academies-list__academy "
              data-academy="true"
              data-id="30015"
            >
              30015
            </div>
            <div
              class="academies-list__academy "
              data-academy="true"
              data-id="30013"
            >
              30013
            </div>
            <div
              class="academies-list__academy "
              data-academy="true"
              data-id="30019"
            >
              30019
            </div>
            <div
              class="academies-list__academy "
              data-academy="true"
              data-id="30021"
            >
              30021
            </div>
            <div
              class="academies-list__academy "
              data-academy="true"
              data-id="30007"
            >
              30007
            </div>
          </div>
        </div>
      </div>
      <div
        class="layout__extra"
      >
        <div
          class="academies-info"
        >
          <div
            class="academy-empty"
          >
            Select an academy to check batteries levels
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App should display an academy info after click on it 1`] = `
<div>
   
  <div
    class="container"
  >
    <div
      class="layout"
    >
      <div
        class="layout__main"
      >
        <div
          class="academies-list"
        >
          <span
            class="header"
          >
            Academies ID:
          </span>
          <div
            class="academies-list__container"
          >
            <div
              class="academies-list__academy "
              data-academy="true"
              data-id="30006"
            >
              30006
            </div>
            <div
              class="academies-list__academy "
              data-academy="true"
              data-id="30015"
            >
              30015
            </div>
            <div
              class="academies-list__academy academies-list__academy_selected"
              data-academy="true"
              data-id="30013"
            >
              30013
            </div>
            <div
              class="academies-list__academy "
              data-academy="true"
              data-id="30019"
            >
              30019
            </div>
            <div
              class="academies-list__academy "
              data-academy="true"
              data-id="30021"
            >
              30021
            </div>
            <div
              class="academies-list__academy "
              data-academy="true"
              data-id="30007"
            >
              30007
            </div>
          </div>
        </div>
      </div>
      <div
        class="layout__extra"
      >
        <div
          class="academies-info"
        >
          <div
            class="academy-info"
          >
            <div
              class="academy-info__devices"
            >
              <span
                class="academy-info__devices-serial"
              >
                Devices serial number (wear level)
              </span>
              <div
                class="academy-info__devices-list"
              >
                <div
                  class="academy-info_device"
                >
                  1805C67HD01952
                  <span
                    class="academy-info_device-level"
                  >
                    (
                    1.08
                    )
                  </span>
                </div>
                <div
                  class="academy-info_device"
                >
                  TT-C67ML-A-0001-01653
                  <span
                    class="academy-info_device-level"
                  >
                    (
                    0.79
                    )
                  </span>
                </div>
                <div
                  class="academy-info_device"
                >
                  NGTT-0842
                  <span
                    class="academy-info_device-level"
                  >
                    (
                    0.74
                    )
                  </span>
                </div>
                <div
                  class="academy-info_device"
                >
                  TT-C67ML-A-0001-01642
                  <span
                    class="academy-info_device-level"
                  >
                    (
                    0.68
                    )
                  </span>
                </div>
                <div
                  class="academy-info_device"
                >
                  TT-C67ML-A-0001-02546
                  <span
                    class="academy-info_device-level"
                  >
                    (
                    0.56
                    )
                  </span>
                </div>
                <div
                  class="academy-info_device"
                >
                  1805C67HD02035
                  <span
                    class="academy-info_device-level"
                  >
                    (
                    0.48
                    )
                  </span>
                </div>
                <div
                  class="academy-info_device"
                >
                  TT-C67ML-A-0001-01556
                  <span
                    class="academy-info_device-level"
                  >
                    (
                    0.45
                    )
                  </span>
                </div>
                <div
                  class="academy-info_device"
                >
                  1805C67HD01991
                  <span
                    class="academy-info_device-level"
                  >
                    (
                    0.42
                    )
                  </span>
                </div>
                <div
                  class="academy-info_device"
                >
                  TT-C67ML-A-0001-01335
                  <span
                    class="academy-info_device-level"
                  >
                    (
                    0.40
                    )
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
